CMakeLists.txt

```diff
@@ -229,7 +229,7 @@ message(STATUS "Desul Atomics support is ${RAJA_ENABLE_DESUL_ATOMICS}")
 if (RAJA_ENABLE_DESUL_ATOMICS)
   # NOTE: ENABLE_OPENMP in desul implies OpenMP OFFLOAD
   if (NOT RAJA_ENABLE_TARGET_OPENMP)
-    set(DESUL_ENABLE_OPENMP Off)
+    set(DESUL_ENABLE_OPENMP Off CACHE BOOL "Disable OpenMP offload atomics to select builtin host atomics")
```

Comment:
@trws how does this play with this? It seems odd to me because RAJA will be configured with ENABLE_OPENMP=On.

https://github.com/trws/desul/blob/desul-config-control/CMakeLists.txt#L26

Reply:
I had to look it up, and test it, to be honest.  The dependent option setup does this:

* if the main variable, ENABLE_OPENMP
   * exists: then the dependent variable exists as a cache variable with
      * if the dependent variable is already defined, the value it has
      * it it isn't, then the default (On) 
   * doesn't exist: then the dependent variable *does not* exist as a cache variable and is created as a local variable
     * if it was already defined, then with the value it has
     * if it isn't, then the undefined default, Off

So, if it's explicitly set before we hit that, the dependent option doesn't override it, it just changes whether it's a treated as a cache variable or not.  If it _isn't_ explicitly set then the values in there take effect.  Not _at all_ obvious.  Seems to work though.

We use the same mechanism to have ENABLE_TESTS on and disable them for say CAMP though, it's something we actually do in a bunch of places it just looks extra odd here because we aren't overriding from the command line.

Reply:
I guess I had only a partial understanding of CMake dependent options. Definitely a delicate dance...  Thank you for your time on this.